js/test/input-mask.test.js

```diff
@@ -0,0 +1,199 @@
+      (spans) => spans[1].textContent
+    );
+    expect(maskedWinKey).toBe("ABCDE-FGHIJ-KLMNO-PQRST-UVWXY");
+  });
+  it("should render cc masked correctly (non Amex)", async () => {
+    const page = await desktopBrowser.newPage();
+    await page.goto(`${config.BASE_URL}/input-mask.php`);
+    await page.type("#cc", "5555555555555555");
+    const maskedWinKey = await page.$$eval(
+      "span.enable-input-mask__mask-pre-val",
+      (spans) => spans[2].textContent
+    );
+    expect(maskedWinKey).toBe("5555 5555 5555 5555");
+  });
+  it("should render cc masked correctly (Amex)", async () => {
+    const page = await desktopBrowser.newPage();
+    await page.goto(`${config.BASE_URL}/input-mask.php`);
+    await page.type("#cc", "3455555555555555");
+    const maskedWinKey = await page.$$eval(
+      "span.enable-input-mask__mask-pre-val",
+      (spans) => spans[2].textContent
+    );
+    expect(maskedWinKey).toBe("3455 555555 55555");
+  });
+
+  //Test keyboard interaction
+  it("should render formatted value correctly post keyboard interaction", async () => {
+    const page = await desktopBrowser.newPage();
+    await page.goto(`${config.BASE_URL}/input-mask.php`, {
+      waitUntil: "domcontentloaded",
+    });
+
+    const telInput = "#tel";
+    await page.focus(telInput);
+    await page.keyboard.type("1234567890");
+    await page.keyboard.press("Backspace");
+    await page.keyboard.press("ArrowLeft");
+    await page.keyboard.type("1");
+    const maskedTelephone = await page.$eval(
+      "span.enable-input-mask__mask-pre-val",
+      (span) => span.textContent
+    );
+    expect(maskedTelephone).toBe("123-456-781");
+  });
+
+  // Test mouse interaction
+  it("should correctly render visual cursor and the input cursor on mouse Interactions", async () => {
+    const page = await desktopBrowser.newPage();
+    await page.goto(`${config.BASE_URL}/input-mask.php`, {
+      waitUntil: "domcontentloaded",
+    });
+
+    const telephoneInput = "#tel";
+
+    // bounding box of input
+    const inputBoundingBox = await page.$eval(telephoneInput, (input) => {
+      const { x, y, width, height } = input.getBoundingClientRect();
+      return { x, y, width, height };
+    });
+
+    // Calculate the middle of the span
+    const middleX = inputBoundingBox.x + inputBoundingBox.width / 2;
+    const middleY = inputBoundingBox.y + inputBoundingBox.height / 2;
+
+    //Initial cursor position, before mouse Interaction
+    const initialCursorPosition = await captureCursorPosition();
+
+    //explicitly move the mouse and click
+    await page.mouse.move(middleX, middleY);
+    await page.mouse.click(middleX, middleY);
+
+    // get the cursor position ofter click, the assumption is after mouse move, an onclick the cursor doesn't Move
+    const cursorPosition = await captureCursorPosition();
+
+    expect(cursorPosition.x).toBeCloseTo(initialCursorPosition.x, 2);
+    expect(cursorPosition.y).toBeCloseTo(initialCursorPosition.y, 2);
```

Comment:
I think the following sets the expectations more clearly, since the initial position is the same as the final position:
```suggestion
    expect(cursorPosition.x).toBe(initialCursorPosition.x);
    expect(cursorPosition.y).toBe(initialCursorPosition.y);
```